pkgs/development/cuda-modules/cuda/overrides.nix

```diff
@@ -187,26 +201,52 @@ filterAndCreateOverrides {
       # backend-stdenv.nix
 
       postPatch =
-        (prevAttrs.postPatch or "")
-        + ''
-          substituteInPlace bin/nvcc.profile \
-            --replace-fail \
-              '$(TOP)/$(_NVVM_BRANCH_)' \
-              "''${!outputBin}/nvvm" \
-            --replace-fail \
-              '$(TOP)/$(_TARGET_DIR_)/include' \
-              "''${!outputDev}/include"
-
-          cat << EOF >> bin/nvcc.profile
-
-          # Fix a compatible backend compiler
-          PATH += "${backendStdenv.cc}/bin":
-
-          # Expose the split-out nvvm
-          LIBRARIES =+ "-L''${!outputBin}/nvvm/lib"
-          INCLUDES =+ "-I''${!outputBin}/nvvm/include"
-          EOF
-        '';
+        let
+          cudaBefore_12_5 =
+            (prevAttrs.postPatch or "")
+            + ''
+              substituteInPlace bin/nvcc.profile \
+                --replace-fail \
+                  '$(TOP)/$(_NVVM_BRANCH_)' \
+                  "''${!outputBin}/nvvm" \
+                --replace-fail \
+                  '$(TOP)/$(_TARGET_DIR_)/include' \
+                  "''${!outputDev}/include"
+
+              cat << EOF >> bin/nvcc.profile
+
+              # Fix a compatible backend compiler
+              PATH += "${backendStdenv.cc}/bin":
+
+              # Expose the split-out nvvm
+              LIBRARIES =+ "-L''${!outputBin}/nvvm/lib"
+              INCLUDES =+ "-I''${!outputBin}/nvvm/include"
+              EOF
+            '';
+          cuda_12_5_andAfter =
+            (prevAttrs.postPatch or "")
+            + ''
+              cat bin/nvcc.profile
+              substituteInPlace bin/nvcc.profile \
+                --replace-fail \
+                  '$(TOP)/$(_TARGET_DIR_)/include' \
+                  "''${!outputDev}/include" \
+                --replace-fail \
+                  '$(TOP)/' \
+                  "''${!outputBin}/"
```

Comment:
iiuc the only difference between these two scripts is the `substituteInPlace` call? in that case could we simplify a bit and just split on that bit?

Reply:
👍🏻 The current let-in version does make it fairly hard to compare the two branches